sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -663,8 +695,54 @@ class Dataset[T] private[sql](
    * @group action
    * @since 1.6.0
    */
+  def show(numRows: Int, truncate: Int): Unit = show(numRows, truncate, extendedMode = false)
+
+  /**
+   * Displays the Dataset in a tabular form. For example:
+   * {{{
+   *   year  month AVG('Adj Close) MAX('Adj Close)
+   *   1980  12    0.503218        0.595103
+   *   1981  01    0.523289        0.570307
+   *   1982  02    0.436504        0.475256
+   *   1983  03    0.410516        0.442194
+   *   1984  04    0.450090        0.483521
+   * }}}
+   *
+   * If `extendedMode` enabled, this command prints a column dat per line:
+   * {{{
+   * -RECORD 0-----------------
+   *  c0  | 0.6988392500990668
+   *  c1  | 0.3035961718851606
+   *  c2  | 0.2446213804275899
+   *  c3  | 0.6132556607194246
+   *  c4  | 0.1904412430355646
+   *  c5  | 0.8856600775630444
+   * -RECORD 1-----------------
+   *  c0  | 0.3942727621020799
+   *  c1  | 0.6501707200059537
+   *  c2  | 0.2550059028276454
+   *  c3  | 0.9806662488156962
+   *  c4  | 0.8533897091838063
+   *  c5  | 0.3911189623246518
+   * -RECORD 2-----------------
+   *  c0  | 0.9024183805969801
+   *  c1  | 0.0242018765375147
+   *  c2  | 0.8508820250344251
+   *  c3  | 0.4593368817024575
+   *  c4  | 0.2216918145613194
+   *  c5  | 0.3756882647319614
+   * }}}
+   *
+   * @param numRows Number of rows to show
+   * @param truncate If set to more than 0, truncates strings to `truncate` characters and
+   *                    all cells will be aligned right.
+   * @param extendedMode Enable expanded table formatting mode to print a column data per line.
```

Comment:
Maybe we can follow what MySQL documents.

@param vertical   Displays the result vertically.